docs/transpile/custom-pass-managers.ipynb

```diff
@@ -0,0 +1,470 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create custom pass managers\n",
+    "\n",
```

Comment:
Can we add an intro here, explaining what a pass manager is, including why you would use one?